kod/object/active/holder/nomoveon/battler/monster/towns/Goad.kod

```diff
@@ -831,11 +829,9 @@ messages:
    "the match, but in general, this will fail if there is no champion, and "
    "will succeed only if there is a challenger spot open."
    {
-      if piCombat_style = STYLE_NO_FIGHT
+      if piCombat_style <> STYLE_ONE_ON_ONE
          {
-           post(self,@ChooseCombat,#style = STYLE_ONE_ON_ONE,#actor=what);
-           %send(self,@saytoOne,#target=what,#message_rsc=Goad_must_choose_style);
-           %return;     
+           Send(self,@ChooseCombat,#style = STYLE_ONE_ON_ONE);    
```

Comment:
Why was this changed from Post to Send?

Reply:
The posting was the cause of the bug this is fixing.  Its double fixed now since the new default is always One_on_one anyhow.

For trigChampion() the posting means we try to say() with a parm=send(poChampion,@getname) when poChampion=$.

For trigChallenge() in the unlikely event that somehow the style isn't one-one-one but there is a champion to challenge this would mean the accept timer doesn't start and I don't know if the fight would even validate and run.  This one is much less likely than the TrigChampion version of the bug - which happens every first fight after a Recreate.  Essentially the Goad has to set the fight to One-on-One in order for everything to function and that has to happen before the rest of the work in each message and so the POST is a mistake.

Now that said - the entire setup is unnecessarily complex because Last-man-standing was cut just prior to production and so we shouldn't really have to set the combat style and check the combat style so often (or at all).  I can implement that type of fix if you think that would be better.  It would mean deleting 3-4 messages and re-writing 4-5 others.

The simplest possible fix to the message bug would be only changing the piCombat_style default property.  That alone does prevent the game update bug - assuming no admin ever changes the piCombat_style property using the admin console.